source/simulator/core.cc

```diff
@@ -118,10 +122,11 @@ namespace aspect
     :
     assemblers (new internal::Assembly::AssemblerLists<dim>()),
     parameters (prm, mpi_communicator_),
+    melt_handler (parameters.include_melt_transport ? new MeltHandler<dim> (prm) : NULL),
     post_signal_creation(
       std_cxx11::bind (&internals::SimulatorSignals::call_connector_functions<dim>,
                        std_cxx11::ref(signals))),
-    introspection (construct_variables<dim>(parameters, signals), parameters),
+    introspection (construct_variables<dim>(parameters, signals, *melt_handler), parameters),
```

Comment:
I think you should just pass the pointer (shared or raw) and then use an `if` in `construct_variables()`.